### PR TITLE
IE8 JScript_DontEnum_Bug

### PR DESCRIPTION
**Pull request by zjruan**
there have some question in my projects, when I run video.js on  IE8.
throw the exception "SCRIPT3: 找不到成员。"

I'll google a lot of information, I finally found this
https://developer.mozilla.org/en-US/docs/ECMAScript_DontEnum_attribute#JScript_DontEnum_Bug

Reproducing  this bug:[run context:ie8]

 var testEle=document.createElement('custom');
 testEle['constructor']=function(){} ;//or testEle.constructor=function(){}

solution:avoid all of the pitfalls

### CODE REVIEW

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: ea27f2ed98b550df068e00ad10a050528b12987c
Build details: https://travis-ci.org/pam/video.js/builds/79115805

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by zjruan** [comment]
I really want to contribute my strength, but I don't know why I build  failed, I just added a Conditional statements

When I  filter the 'constructor ',it  run success

**Comment by gkatsev** [comment]
The build failed because of linting errors, see the output here: https://travis-ci.org/videojs/video.js/builds/79115792#L120-L126
Also, make sure to run `grunt` locally to have the build pass locally.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 59eb69844029a2ec198d2e1aad6338f5ebb04433
Build details: https://travis-ci.org/pam/video.js/builds/79148169

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by pam** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: ffd7fee6fa5d07d3b4e70da0d2af86d6af4fe3f8
Build details: https://travis-ci.org/pam/video.js/builds/79148185

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: src/js/tracks/text-track-cue-list.js [hunk 1]
```diff
@@ -21,7 +21,9 @@ let TextTrackCueList = function(cues) {
     list = document.createElement('custom');
 
     for (let prop in TextTrackCueList.prototype) {
-      list[prop] = TextTrackCueList.prototype[prop];
+      if(prop!=='constructor'){
```
Comment (gkatsev) [inline src/js/tracks/text-track-cue-list.js]: Some style updates:
* space after `if`
* space around operator (`!==`)
* space before opening `{`

Also below.

**Comment by gkatsev** [comment]
Looks good, other than a minor stylistic issue. If you can update it, I'll be happy to merge it in.
Thanks!

**Comment by pam** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 33c6293276e1640884ac71318c20e781ca4a8974
Build details: https://travis-ci.org/pam/video.js/builds/90227162

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by gkatsev** [comment]
Awesome, thanks. I'll get that merged in tomorrow.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: ed2363b040c01cd873210e5f172c261173740845
Build details: https://travis-ci.org/pam/video.js/builds/90227108

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by zjruan** [comment]
I feel very honored  ^_^

**Comment by dmlap** [comment]
LGTM

**Pull request closed without merging**